### ISSUE TITLE
Problem with selecting arcade dat file on switch version of retroarch

### ISSUE THREAD

**Issue by chikenstew** [libretro/RetroArch#13265]
I'm trying to do import content>manual scan on the latest switch version of retroarch (1.9.13), but whenever I select an arcade dat file, it always gives me "dat file too large insufficient free memory" error no matter which dat file I select.

**Comment by icescoop12** [comment]
has this been fixed or is there a solution?

**Comment by ChrysMacZ** [comment]
I'm think not, same problem for me on TvOs

**Comment by zpaolo11x** [comment]
Same issue on ipadOS, now that RA is available to all non-jailbroken ipads this is becoming a major issue...

**Comment by zpaolo11x** [comment]
> has this been fixed or is there a solution?

There is a workaround though: if you open the xml dat file in a text editor you'll notice that for each <game> tag there's a series of <rom> tags, they are listing the file inside each mame .zip archive. If your roms are in zip format all these entries are not needed and you can bulk remove those lines, reducing the size of the file to a couple of megabytes, that RA will happily read and use to apply the game names.

**Comment by saghul** [comment]
The directory scan is working for me, just the NeoGeo ROMs were missing. I then made a manual scan with the NeoGeo only DAT file (it's 2.5MB) and that did the trick.

**Comment by zpaolo11x** [comment]
> The directory scan is working for me, just the NeoGeo ROMs were missing. I then made a manual scan with the NeoGeo only DAT file (it's 2.5MB) and that did the trick.

It's working because you have a dat file of just 2.5MB, the one for fbNeo is 11 to 12 MB and that's causing the issue with manual scan, unless you strip the dat of all the unused <rom> tags and reduce its size.

**Comment by saghul** [comment]
Correct. My point was that (for me at least) the non-manual scan worked for fbneo, _excep_ for the NeoGeo games, and manually scanning just those is fine because the file is indeed smaller.
